phpBB/phpbb/install/controller/archive_download.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\install\controller;
+
+use phpbb\install\helper\config;
+use Symfony\Component\HttpFoundation\Response;
+
+class archive_download
+{
+	/**
+	 * @var config
+	 */
+	protected $installer_config;
+
+	/**
+	 * Constructor
+	 *
+	 * @param config $config
+	 */
+	public function __construct(config $config)
+	{
+		$this->installer_config = $config;
+		$this->installer_config->load_config();
+	}
+
+	/**
+	 * Sends response with the merge conflict archive
+	 *
+	 * @return Response
+	 */
+	public function conflict_archive()
+	{
+		$filename = $this->installer_config->get('update_file_conflict_archive', false);
+
+		if (!$filename)
+		{
+			die ('The requested file is not exist.');
+		}
+
+		return $this->send_response($filename);
+	}
+
+	/**
+	 * Sends response with the updated files' archive
+	 *
+	 * @return Response
+	 */
+	public function update_archive()
+	{
+		$filename = $this->installer_config->get('update_file_archive', '');
+
+		if (!$filename)
+		{
+			die ('The requested file is not exist.');
+		}
+
+		return $this->send_response($filename);
+	}
+
+	/**
+	 * Generates a download response
+	 *
+	 * @param string	$filename	Path to the file to download
+	 *
+	 * @return Response	Response object
+	 */
+	private function send_response($filename)
+	{
+		$response = new Response();
+
+		$response->headers->set('Cache-Control', 'private');
+		$response->headers->set('Content-type', mime_content_type($filename));
+		$response->headers->set('Content-Disposition', 'attachment; filename="' . basename($filename) . '";');
+		$response->headers->set('Content-length', filesize($filename));
+
+		$response->sendHeaders();
+		$response->setContent(file_get_contents($filename));
+
+		return $response;
```

Comment:
you can use a `BinaryFileResponse` here

http://api.symfony.com/2.7/Symfony/Component/HttpFoundation/BinaryFileResponse.html
https://symfony.com/doc/current/components/http_foundation/introduction.html#serving-files

On a side note, you should never call `sendHeaders()` manually, it is done automatically when `send()` is called